### DISCUSSION TITLE
How to setup complexity?

### DISCUSSION THREAD

**Discussion by Diluka** [TriPSs/nestjs-query#141] [Q&A] [answered]
according to https://docs.nestjs.com/graphql/complexity

**Comment by TriPSs** [comment]
https://tripss.github.io/nestjs-query/docs/faq/#does-nestjs-query-support-specifying-complexity

**Reply by Diluka** [reply]
```ts
@Query({ complexity: (options: ComplexityEstimatorArgs) => options.args.count * options.childComplexity })
items(@Args('count') count: number) {
  return this.itemsService.getItems({ count });
}
```

how to set complexity to generated operations like this? `limit * childComplexity`

**Reply by TriPSs** [reply]
They are mapped directly to the nestjs `@Query` so you should be able to do the same.

**Reply by Diluka** [reply]
```ts
@Module({
  imports: [
    TypeOrmModule.forFeature([User]),
    NestjsQueryGraphQLModule.forFeature({
      imports: [NestjsQueryTypeOrmModule.forFeature([User])],
      resolvers: [
        {
          EntityClass: User,
          DTOClass: UserDto,
          read: {  },
          enableTotalCount: true,
          pagingStrategy: PagingStrategies.OFFSET,
        },
      ],
    }),
  ],
})
export class UserModule {}
```

I mean how to add the config here? I want to complexity to be `paging.limit * childComplexity`

**Reply by TriPSs** [reply]
Sorry now I understand, I see what you mean and after checking I saw we do not support this yet, think this would be a nice addition that next to `read.description` we also add `read.complexity` etc.

**Reply by TriPSs** [reply, marked as answer]
FYI: This has been added here by #142
